test/test_utils/__init__.py

```diff
@@ -173,6 +173,15 @@ def is_benchmark_dev_context():
     return ENABLE_BENCHMARK_DEV_MODE
 
 
+def is_time_for_canary_safety_scan():
+    """
+    Canary tests erun every 15 minutes. 
+    Using a 20 minutes interval to make tests run only once a day around 10 am PST.
+    """
+    current_utc_time = time.gmtime()
+    return current_utc_time.tm_hour == 16 and (0 < current_utc_time.tm_min < 20)
```

Comment:
why 16

Reply:
The code will be much more readable if you were to use https://pypi.org/project/pytz/ or other Time Zone management packages along with `datetime` to accurately check/show the datetime.

Reply:
It's UTC time so it will be 9 am PST and 10 am winter time.

Reply:
Using a new library only to convert time from UTC to PST looks like an overkill to me.